lib/msal-common/src/cache/CacheHelpers.ts

```diff
@@ -56,114 +56,121 @@ export class CacheHelpers {
+        });
+        const foundTokens: Array<Promise<AccessTokenCacheItem>> = tokenKeys.map(async (key) => {
+            const valueAtKey = await this.cacheStorage.getItem(key);
+            if (valueAtKey) {
+                try {
+                    const parsedAccessTokenKey: AccessTokenKey = JSON.parse(key) as AccessTokenKey;
+                    if (this.checkForExactKeyMatch(parsedAccessTokenKey, clientId, authority, resource, homeAccountIdentifier)) {
+                        const newAccessTokenCacheItem = new AccessTokenCacheItem(parsedAccessTokenKey, JSON.parse(valueAtKey) as AccessTokenValue);
+                        return newAccessTokenCacheItem;
                     }
+                } catch (e) {
+                    throw ClientAuthError.createCacheParseError(key);
                 }
             }
-            return tokens;
-        }, []);
+            return null;
+        });
 
-        return results;
+        return Promise.all(foundTokens).then(arr => arr.filter(item => item));
     }
 
     /**
      * Remove all access tokens in the cache
      * @param clientId
      * @param homeAccountIdentifier
      */
-    removeAllAccessTokens(clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): void {
-        this.cacheStorage.getKeys().forEach((key) => {
-            const keyMatches = key.match(clientId) && key.match(authority) && key.match(resource) && key.match(homeAccountIdentifier);
-            if (keyMatches) {
-                try {
-                    const parseAtKey = JSON.parse(key) as AccessTokenKey;
-                    if (this.checkForExactKeyMatch(parseAtKey, clientId, authority, resource, homeAccountIdentifier)) {
-                        this.cacheStorage.removeItem(key);
-                    }
-                } catch (e) {
-                    throw ClientAuthError.createCacheParseError(key);
+    async removeAllAccessTokens(clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): Promise<void> {
+        const cacheKeys = await this.cacheStorage.getKeys();
+        const tokenKeys: Array<string> = cacheKeys.filter((key) => {
+            return key.match(clientId) && key.match(authority) && key.match(resource) && key.match(homeAccountIdentifier);
+        });
+        const removedTokens: Array<Promise<void>> = tokenKeys.map(async (key) => {
+            try {
+                const parsedAccessTokenKey = JSON.parse(key) as AccessTokenKey;
+                if (this.checkForExactKeyMatch(parsedAccessTokenKey, clientId, authority, resource, homeAccountIdentifier)) {
+                    return await this.cacheStorage.removeItem(key);
                 }
+                return null;
+            } catch (e) {
+                throw ClientAuthError.createCacheParseError(key);
             }
-        });
+        }).filter(resolution => resolution);
+        return Promise.all(removedTokens).then(() => null);
```

Comment:
Since this returns Promise of void, there is no filter needed, just wait for all to resolve then return null